src/c++/library/http_client.cc

```diff
@@ -1781,7 +2082,14 @@ InferenceServerHttpClient::PreRunProcessing(
 
   // Add the buffers holding input tensor data
   for (const auto this_input : inputs) {
-    if (!this_input->IsSharedMemory()) {
+    bool json_data{false};
+    if (std::shared_ptr<InferInput::HttpOptions> options =
+            std::dynamic_pointer_cast<InferInput::HttpOptions>(
+                this_input->GetOptions())) {
+      json_data = options->json_data;
```

Comment:
Why do you need to make a copy of the shared pointer?

Reply:
Irrelevant now 👍